doc/sphinx/proof-engine/ltac.rst

```diff
@@ -713,100 +747,183 @@ tactic
-   by substituting :n:`v__i` to each occurrence of :n:`@ident__i`, for
-   i = 1, ..., n. There are no dependencies between the :n:`@ltac_expr__i` and the
-   :n:`@ident__i`.
+   Evaluates each :n:`@let_clause`, substitutes the bound variables into
+   :n:`@ltac_expr` and then evaluates :n:`@ltac_expr`.  There are no dependencies
+   between the :n:`@let_clause`\s.
 
-   Local definitions can be made recursive by using :n:`let rec` instead of :n:`let`.
-   In this latter case, the definitions are evaluated lazily so that the rec
-   keyword can be used also in non-recursive cases so as to avoid the eager
+   Creates a local definition.
+   Using :tacn:`let` `rec` makes the definition recursive and causes
+   the definitions to be evaluated lazily.  Thus, :tacn:`let` `rec`
+   can be used in non-recursive cases to avoid eager
    evaluation of local definitions.
 
    .. but rec changes the binding!!
 
+   :tacn:`let` is a :token:`binder_tactic`.
+
 Application
 ~~~~~~~~~~~
 
 An application is an expression of the following form:
 
-.. tacn:: @qualid {+ @tacarg}
+.. tacn:: @qualid {* @tactic_arg }
 
    The reference :n:`@qualid` must be bound to some defined tactic definition
-   expecting at least as many arguments as the provided :n:`tacarg`. The
+   expecting at least as many arguments as the provided :n:`tactic_arg`. The
    expressions :n:`@ltac_expr__i` are evaluated to :n:`v__i`, for i = 1, ..., n.
 
    .. what expressions ??
 
-Function construction
-~~~~~~~~~~~~~~~~~~~~~
+Function construction: fun
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 A parameterized tactic can be built anonymously (without resorting to
 local definitions) with:
 
-.. tacn:: fun {+ @ident} => @ltac_expr
+.. tacn:: fun {+ @name } => @ltac_expr
 
    Indeed, local definitions of functions are a syntactic sugar for binding
    a :n:`fun` tactic to an identifier.
 
-Pattern matching on terms
-~~~~~~~~~~~~~~~~~~~~~~~~~
+   :tacn:`fun` is a :token:`binder_tactic`.
+
+Pattern matching on terms: match
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tacn:: @match_key @ltac_expr with {? %| } {+| @match_rule } end
+   :name: match; lazymatch; multimatch
+
+   .. insertprodn match_key cpattern
+
+   .. prodn::
+      match_key ::= match
+      | multimatch
+      | lazymatch
+      match_rule ::= {| @match_pattern | _ } => @ltac_expr
+      match_pattern ::= @cpattern
+      | context {? @ident } [ @cpattern ]
+      cpattern ::= @term
+
+   .. todo I don't fully understand these descriptions.  I think some concepts
+      such as backtracking need to be explained first and that the lanugage
+      can probably be simplified.
+
+   :tacn:`match`, :tacn:`lazymatch` and :tacn:`multimatch` are :token:`ltac_expr1`\s.
+
+   `multimatch`
+      Allows subsequent tactics to
+      backtrack into a right-hand side tactic which has backtracking points
+      left and trigger the selection of a new matching branch when all the
```

Comment:
Yes, "left over" is a reasonable alternative.  But I can just refer to cpattern and ltac_expr instead of right and left hand sides, that seems clearer and less wordy.